components/batch-system/src/router.rs

```diff
@@ -220,7 +220,13 @@ where
                     .unwrap()
                     .force_send(m, &self.normal_scheduler)
             }
-            Err(TrySendError::Disconnected(m)) => Err(SendError(m)),
+            Err(TrySendError::Disconnected(m)) => {
+                if self.is_shutdown() {
```

Comment:
I don't think it's a good idea. Only caller knows what's the most appropriate approach to handle shutdown.

Reply:
Maybe it's general enough. I can't imagine a case that the caller needs to know this send error during the shutdown.

Reply:
It's the problem of API design choice. The change assumes callers are always OK to drop messages without reasoning the error type, which may not be the case for future. For example, if callers need to clean up some resources for error, while holding resource when succeed.

If it's a generic case, then it should be implemented as a different API and give it a clear name.

Reply:
@Little-Wallace I don't see it's commented or adjusted.  :)

Reply:
If this router has shutdown, it means that all of the message shall be ignore, because there is no thread to deal with these messages.

Reply:
Sorry for I missing your comment.... I think we can assume that when the router is `shut_down`, all the message has been send successfully but there is no thread to deal with them. The following two logics is equal:
```
thread1: send(msg).is_ok();
thread2: recv(msg).is_ok(); if is_shutdown() {  drop(msg);  exit(); }
```

```
thread1: send(msg).is_err();
thread2: if is_shutdown() { exit(); }
```

Reply:
It means all of the messages are not being handled by the threads, doesn't necessary mean to be ignored or reported as error.

Reply:
If shutdown occurs,  TiKV do not need to care about whether the messages have send successfully.

Reply:
> The following two logics is equal:

Actually, TiKV doesn't just drop all messages unconditionally. It will invoke the callbacks if there are any.

> If shutdown occurs, TiKV do not need to care about whether the messages have send successfully.

Like I said above, "It's the problem of API design choice. The change assumes callers are always OK to drop messages without reasoning the error type, which may not be the case for future. For example, if callers need to clean up some resources for error, while holding resource when succeed.". Or maybe someone just tries to use `force_send` to send a command.